### PR TITLE
feat: gpkg grid validation

### PR DESCRIPTION
**Pull request by cl-shaharshavit-rnd**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔/✖                                                                        |
| New feature     | ✔/✖                                                                        |
| Breaking change | ✔/✖                                                                        |
| Deprecations    | ✔/✖                                                                        |
| Documentation   | ✔/✖                                                                        |
| Tests added     | ✔/✖                                                                        |
| Chore            | ✔/✖                                                                       |

Related issues: #XXX , #XXX ...
Closes #XXX ...

Further  information:

### CODE REVIEW

File: MergerLogic/Utils/GpkgUtils.cs [hunk 1]
```diff
@@ -681,5 +682,66 @@ public void UpdateTileMatrixTable(bool isOneXOne = false)
...
+                    string maxY = isOneXOne ? "180" : "90";
+                    command.CommandText = "SELECT count(*) FROM gpkg_tile_matrix_set " +
+                                          $"WHERE table_name = '{this._tileCache}' AND srs_id = {Utils.GeoUtils.SRID} " +
+                                          "AND min_x = -180 AND max_x = 180 " +
+                                          $"AND min_y = -{maxY} AND max_y = {maxY}";
+                    long count = (long)command.ExecuteScalar();
+                    var isValid = count == 1;
+                    if (!isValid)
+                    {
+                        return false;
```
Comment (asafMasa) [inline MergerLogic/Utils/GpkgUtils.cs:702]: add log on failed check

File: MergerLogic/Utils/GpkgUtils.cs [hunk 2]
```diff
@@ -681,5 +682,66 @@ public void UpdateTileMatrixTable(bool isOneXOne = false)
...
+                                xTiles <<= 1;
+                                zoom++;
+                            }
+
+                            if (reader.GetInt32(1) != xTiles || reader.GetInt32(2) != yTiles ||
+                                reader.GetInt32(3) != tileSize || reader.GetInt32(4) != tileSize ||
+                                !reader.GetDouble(5).IsApproximatelyEqualTo(res, doublePrecession) ||
+                                !reader.GetDouble(6).IsApproximatelyEqualTo(res, doublePrecession))
+                            {
+                                return false;
```
Comment (asafMasa) [inline MergerLogic/Utils/GpkgUtils.cs:737]: add log on failed check (table, zoom, expected fields vs actual fields)

**Review by asafMasa — CHANGES_REQUESTED** [review]
💯

File: MergerLogic/Extensions/DoubleExtenstion.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿namespace MergerLogic.Extensions
+{
+    public static class DoubleExtenstion
+    {
+        public static bool IsApproximatelyEqualTo(this double initialValue, double value, double maximumDifferenceAllowed)
+        {
+            // Handle comparisons of floating point values that may not be exactly the same
+            return (Math.Abs(initialValue - value) < maximumDifferenceAllowed);
```
[the remark this answers is not in the document]
Comment (cl-shaharshavit-rnd) [reply MergerLogic/Extensions/DoubleExtenstion.cs:8]: doesn't really matters in that case.
Updated anyways

**Review by asafMasa — APPROVED** [review]

**Review by asafMasa — APPROVED** [review]

**Pull request merged**